src/main/java/com/example/shopproject/entity/Role.java

```diff
@@ -1,2 +1,57 @@
-package com.example.shopproject.entity;public class Role {
+package com.example.shopproject.entity;
+
+import org.springframework.security.core.GrantedAuthority;
+
+import javax.persistence.*;
+import java.util.Set;
+
+@Entity
+@Table(name = "t_role")
+public class Role implements GrantedAuthority {
+    @Id
+    private Long id;
+    private String name;
+    @Transient
+    @ManyToMany(mappedBy = "roles")
+    private Set<User> users;
```

Comment:
better to initialize collections, and use addUsers method instead of setter, I think `@Data` annotation from lombok will help you